core/src/test/java/org/elasticsearch/search/aggregations/pipeline/PercentilesBucketIT.java

```diff
@@ -0,0 +1,527 @@
+        List<Terms.Bucket> buckets = terms.getBuckets();
+        assertThat(buckets.size(), equalTo(0));
+
+        PercentilesBucket percentilesBucketValue = response.getAggregations().get("percentiles_bucket");
+        assertThat(percentilesBucketValue, notNullValue());
+        assertThat(percentilesBucketValue.getName(), equalTo("percentiles_bucket"));
+
+        try {
+            percentilesBucketValue.percentile(2.0);
+            fail("2.0 was not a valid percent, should have thrown exception");
+        } catch (IllegalArgumentException exception) {
+            // All good
+        } catch (Exception exception) {
+            fail("Unknown exception caught: " + exception);
+        }
+    }
+
+    @Test
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/13179")
+    public void testBadPercents() throws Exception {
+        Double[] badPercents = {-1.0, 110.0};
+
+        try {
+            SearchResponse response = client().prepareSearch("idx")
+                    .addAggregation(terms("terms").field("tag").subAggregation(sum("sum").field(SINGLE_VALUED_FIELD_NAME)))
+                    .addAggregation(percentilesBucket("percentiles_bucket")
+                            .setBucketsPaths("terms>sum")
+                            .percents(badPercents)).execute().actionGet();
+
+            assertSearchResponse(response);
+
+            Terms terms = response.getAggregations().get("terms");
+            assertThat(terms, notNullValue());
+            assertThat(terms.getName(), equalTo("terms"));
+            List<Terms.Bucket> buckets = terms.getBuckets();
+            assertThat(buckets.size(), equalTo(0));
+
+            PercentilesBucket percentilesBucketValue = response.getAggregations().get("percentiles_bucket");
+
+            fail("Illegal percent's were provided but no exception was thrown.");
+        } catch (IllegalStateException exception) {
+            // All good
+        } catch (Exception exception) {
+            fail("unknown exception caught: " + exception);
+        }
+
+    }
+
+    @Test
+    public void testNested() throws Exception {
+        SearchResponse response = client()
+                .prepareSearch("idx")
+                .addAggregation(
+                        terms("terms")
+                                .field("tag")
+                                .order(Terms.Order.term(true))
+                                .subAggregation(
+                                        histogram("histo").field(SINGLE_VALUED_FIELD_NAME).interval(interval)
+                                                .extendedBounds((long) minRandomValue, (long) maxRandomValue))
+                                .subAggregation(sumBucket("sum_histo_bucket").setBucketsPaths("histo>_count")))
+                .addAggregation(percentilesBucket("percentile_terms_bucket")
+                        .setBucketsPaths("terms>sum_histo_bucket")
+                        .percents(PERCENTS)).execute().actionGet();
```

Comment:
Could we either do the percentiles of percentiles here or have another test that calculates the max bucket of a terms agg which is calculating percentiles over a histogram?  This way it would check that the agg can both reference an inner sibling pipeline agg and be referenced by an out sibling pipeline agg.